src/tachimawari_dynamixel/dynamixel.cpp

```diff
@@ -0,0 +1,387 @@
+    getdata_state = group_sync_read.isAvailable(
+      joints.at(index).get_id(), static_cast<uint8_t>(start_address), data_length);
+    if (getdata_state != true) {
+      RCLCPP_ERROR_STREAM(
+        get_logger(), "[ID:" << std::setfill('0') << std::setw(2) <<
+          static_cast<int>(joints.at(index).get_id()) << "]. syncread getdata failed");
+    } else {
+      // Get Dynamixel present position value
+      data_result = group_sync_read.getData(
+        joints.at(index).get_id(), static_cast<uint8_t>(start_address), data_length);
+      joints.at(index).set_present_position(data_result);
+    }
+  }
+
+  // Clear syncread parameter storage
+  group_sync_read.clearParam();
+
+  return true;
+}
+
+bool Dynamixel::bulk_read_joints(
+  std::vector<Joint> & joints, MXAddress start_address,
+  int data_length)
+{
+  // Initialize GroupBulkRead instance
+  dynamixel::GroupBulkRead group_bulk_read(port_handler, packet_handler);
+
+  // Add parameter storage for Dynamixel data
+  bool addparam_state = false;
+  for (auto joint : joints) {
+    addparam_state = group_bulk_read.addParam(
+      joint.get_id(), static_cast<uint16_t>(start_address), data_length);
+    if (addparam_state != true) {
+      RCLCPP_ERROR_STREAM(
+        get_logger(), "[ID:" << std::setfill('0') << std::setw(2) <<
+          static_cast<int>(joint.get_id()) << "]. bulkread addparam failed");
+      return false;
+    }
+  }
+
+  // Syncread data
+  int comm_result = COMM_TX_FAIL;
+  comm_result = group_bulk_read.txRxPacket();
+  if (comm_result != COMM_SUCCESS) {
+    RCLCPP_ERROR_STREAM(
+      get_logger(), "failed to bulkread data " << packet_handler->getTxRxResult(comm_result));
+    return false;
+  }
+
+  bool getdata_state = false;
+  int32_t data_result = 0;
+
+  for (int index = 0; index < static_cast<int>(joints.size()); index++) {
+    // Check if groupsyncread data of Dynamixel is available
+    getdata_state = group_bulk_read.isAvailable(
+      joints.at(index).get_id(), static_cast<uint8_t>(start_address), data_length);
+    if (getdata_state != true) {
+      RCLCPP_ERROR_STREAM(
+        get_logger(), "[ID:" << std::setfill('0') << std::setw(2) <<
+          static_cast<int>(joints.at(index).get_id()) << "]. bulkread getdata failed");
+    } else {
+      // Get Dynamixel present position value
+      data_result = group_bulk_read.getData(
+        joints.at(index).get_id(), static_cast<uint8_t>(start_address), data_length);
+      joints.at(index).set_present_position(data_result);
+    }
+  }
+
+  // Clear syncread parameter storage
+  group_bulk_read.clearParam();
+
+  return true;
+}
+
+bool Dynamixel::init_joints_present_position(std::vector<Joint> joints)
+{
+  bool init_state = true;
+
+  if (!init_joints_state) {
+    init_state = sync_read_joints(joints);
+    this->joints = joints;
+    init_joints_state = true;
+  } else {
+    for (int index = 0; index < static_cast<int>(joints.size()); index++) {
```

Comment:
Use `size_t` data type and remove `static_cast`. Do this to the other for loop too.